pkg/printers/template_flags.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// TemplatePrintFlags provides default flags necessary for template printing.
+// Given the following flag values, a printer can be requested that knows
+// how to handle printing based on these values.
+type TemplatePrintFlags struct {
+	// indicates if it is OK to ignore missing keys for rendering
+	// an output template.
+	AllowMissingKeys *bool
+	TemplateArgument *string
+}
+
+func (f *TemplatePrintFlags) Validate() error {
+	if f.TemplateArgument == nil || len(*f.TemplateArgument) == 0 {
+		return fmt.Errorf("missing --template value")
+	}
+	return nil
+}
+
+func (f *TemplatePrintFlags) ToPrinter(outputFormat string) (ResourcePrinter, error) {
```

Comment:
Update to accept the **actual** outputFormat and reconcile against flags: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/printing.go#L201-L222